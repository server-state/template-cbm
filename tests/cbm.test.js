import React from 'react';
import { create } from "react-test-renderer";
import CBM from "../src";


test('two plus two is four', () => {
    expect(2 + 2).toBe(4);
});

describe("CBM specification tests", () => {
    it("should contain all necessary infos", () => {
        expect(CBM.info).toHaveProperty("name");
    });

    it("should render without any complications", () => {
        console.error = jest.fn();
        const mockData = "Test data";
        const cbm = create(
            <CBM.component
                data={mockData}
                onRegisterAction={() => {}}
                onRefresh={() => {}}
            />
        );

        expect(cbm).toMatchSnapshot();
    });
});