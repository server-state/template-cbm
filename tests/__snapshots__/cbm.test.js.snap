// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CBM specification tests should render without any complications 1`] = `
<div>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    My CBM module!
  </p>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    Data: 
    Test data
  </p>
</div>
`;
